[Automation] native-image run fails for io.netty:netty-codec-socks:4.1.79.Final

## native-image run on ubuntu-latest / JDK 25 / mode current-defaults

Failure kind: native-image run
OS: ubuntu-latest
JDK: 25
Native-image mode: current-defaults
Reproducer: `GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="4.1.79.Final" ./gradlew clean test -Pcoordinates="io.netty:netty-codec-socks:4.1.60.Final"`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/25248087844/job/74035700675
Last 300 lines of the log (excerpt truncated to fit the GitHub issue size limit):
```console
e/graal/blob/master/docs/reference-manual/native-image/BuildOutput.md
------------------------------------------------------------------------------------------------------------------------
[1/8] Initializing...                                                                                    (6.2s @ 0.18GB)
 Java version: 25.0.3+9-LTS, vendor version: Oracle GraalVM 25.0.3+9.1
 Graal compiler: optimization level: b, target machine: x86-64-v3, PGO: off
 C compiler: gcc (linux, x86_64, 13.3.0)
 Garbage collector: Serial GC (max heap size: 80% of RAM)
 2 user-specific feature(s):
 - com.oracle.svm.thirdparty.gson.GsonFeature
 - org.graalvm.junit.platform.JUnitPlatformFeature
------------------------------------------------------------------------------------------------------------------------
 1 experimental option(s) unlocked:
 - '-H:BuildOutputJSONFile' (origin(s): command line)
------------------------------------------------------------------------------------------------------------------------
Build resources:
 - 12.67GB of memory (75.6% of system memory, $CI set to 'true')
 - 4 thread(s) (100.0% of 4 available processor(s), determined at start)
[junit-platform-native] Running in 'test listener' mode using files matching pattern [junit-platform-unique-ids*] found in folder [/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.netty/netty-codec-socks/4.1.60.Final/build/test-results/test/testlist] and its subfolders.
[2/8] Performing analysis...  [*****]                                                                   (36.0s @ 0.89GB)
    8,566 types,   9,778 fields, and  38,213 methods found reachable
    2,913 types,     297 fields, and   2,703 methods registered for reflection
       60 types,      68 fields, and      52 methods registered for JNI access
        0 downcalls and 0 upcalls registered for foreign access
        4 native libraries: dl, pthread, rt, z
[3/8] Building universe...                                                                               (4.3s @ 0.98GB)
[4/8] Parsing methods...      [**]                                                                       (4.3s @ 0.63GB)
[5/8] Inlining methods...     [***]                                                                      (2.4s @ 0.70GB)
[6/8] Compiling methods...    [****]                                                                    (19.4s @ 0.82GB)
[7/8] Laying out methods...   [**]                                                                       (3.4s @ 0.79GB)
[8/8] Creating image...       [**]                                                                       (3.4s @ 0.90GB)
   9.85MB (35.49%) for code area:    23,641 compilation units
  14.02MB (50.54%) for image heap:  214,914 objects and 155 resources
   3.88MB (13.97%) for other data
  27.75MB in total image size, 25.63MB in total file size
------------------------------------------------------------------------------------------------------------------------
Top 10 origins of code area:                                Top 10 object types in image heap:
   4.54MB java.base                                            2.88MB byte[] for code metadata
   2.14MB svm.jar (Native Image)                               2.63MB byte[] for java.lang.String
 563.99kB assertj-core-3.22.0.jar                              1.39MB java.lang.String
 481.09kB java.xml                                             1.08MB java.lang.Class
 270.35kB netty-common-4.1.79.Final.jar                      912.54kB byte[] for general heap data
 199.61kB netty-buffer-4.1.79.Final.jar                      890.86kB com.oracle.svm.core.hub.DynamicHubCompanion
 198.00kB junit-jupiter-engine-5.9.2.jar                     547.68kB heap alignment
 120.81kB junit-platform-launcher-1.9.2.jar                  449.86kB java.util.concurrent.ConcurrentHashMap$Node
 103.55kB java.logging                                       300.61kB java.util.HashMap$Node
  99.49kB junit-platform-engine-1.9.2.jar                    281.26kB java.lang.Object[]
 756.58kB for 30 more packages                                 2.66MB for 2050 more object types
                            Use '--emit build-report' to create a report with more details.
------------------------------------------------------------------------------------------------------------------------
Security report:
 - Binary includes Java deserialization.
 - CycloneDX SBOM with 22 component(s) is embedded in binary (1.06kB). 1 type(s) could not be associated to a component.
 - Advanced obfuscation not enabled; enable with '-H:AdvancedObfuscation=""' (experimental support).
------------------------------------------------------------------------------------------------------------------------
Recommendations:
 G1GC: Use the G1 GC ('--gc=G1') for improved latency and throughput.
 PGO:  Use Profile-Guided Optimizations ('--pgo') for improved throughput.
 FUTR: Use '--future-defaults=all' to prepare for future releases.
 HEAP: Set max heap for improved and more predictable memory usage.
 CPU:  Enable more CPU features with '-march=native' for improved performance.
------------------------------------------------------------------------------------------------------------------------
                        4.9s (5.9% of total time) in 704 GCs | Peak RSS: 1.72GB | CPU load: 3.63
------------------------------------------------------------------------------------------------------------------------
Build artifacts:
 /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.netty/netty-codec-socks/4.1.60.Final/build/native/nativeTestCompile/io.netty.netty-codec-socks_tests-tests (executable)
 /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.netty/netty-codec-socks/4.1.60.Final/build/native/nativeTestCompile/libmanagement_ext.so (jdk_library)
 /tmp/native-image-build-output.json (build_info)
========================================================================================================================
Finished generating 'io.netty.netty-codec-socks_tests-tests' in 1m 21s.
[native-image-plugin] Native Image written to: /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.netty/netty-codec-socks/4.1.60.Final/build/native/nativeTestCompile
WARNING: A terminally deprecated method in sun.misc.Unsafe has been called
WARNING: sun.misc.Unsafe::getLong has been called by io.netty.util.internal.PlatformDependent0$4 (file:/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.netty/netty-codec-socks/4.1.60.Final/build/native/nativeTestCompile/io.netty.netty-codec-socks_tests-tests)
WARNING: Please consider reporting this to the maintainers of class io.netty.util.internal.PlatformDependent0$4
WARNING: sun.misc.Unsafe::getLong will be removed in a future release

> Task :nativeTest FAILED
[junit-platform-native] Running in 'test listener' mode using files matching pattern [junit-platform-unique-ids*] found in folder [/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.netty/netty-codec-socks/4.1.60.Final/build/test-results/test/testlist] and its subfolders.
JUnit Platform on Native Image - report
----------------------------------------

io_netty.netty_codec_socks.Netty_codec_socksTest > portUnificationServerHandlerSelectsSocks4AndSocks5Decoders() FAILED

io_netty.netty_codec_socks.Netty_codec_socksTest > socks4aCommandRequestSupportsDomainNames() FAILED

io_netty.netty_codec_socks.Netty_codec_socksTest > protocolConstantsAndExtensibleValueObjectsExposeWireValues() SUCCESSFUL

io_netty.netty_codec_socks.Netty_codec_socksTest > replayingDecodersWaitForCompleteFramesBeforeEmittingMessages() FAILED

io_netty.netty_codec_socks.Netty_codec_socksTest > socks5CommandCodecsUseCustomAddressCodecs() FAILED

io_netty.netty_codec_socks.Netty_codec_socksTest > socks5CommandCodecsSupportIpv4DomainAndIpv6Addresses() FAILED

io_netty.netty_codec_socks.Netty_codec_socksTest > socks5InitialNegotiationAndPasswordAuthenticationRoundTrip() FAILED

io_netty.netty_codec_socks.Netty_codec_socksTest > socks4MessagesRoundTripThroughClientAndServerCodecs() FAILED


Failures (7):
  JUnit Jupiter:Netty_codec_socksTest:portUnificationServerHandlerSelectsSocks4AndSocks5Decoders()
    MethodSource [className = 'io_netty.netty_codec_socks.Netty_codec_socksTest', methodName = 'portUnificationServerHandlerSelectsSocks4AndSocks5Decoders', methodParameterTypes = '']
    => java.lang.ExceptionInInitializerError
       java.base@25.0.3/java.lang.Class.ensureInitialized(DynamicHub.java:778)
       io.netty.buffer.ByteBufUtil.<clinit>(ByteBufUtil.java:86)
       io.netty.buffer.ByteBufAllocator.<clinit>(ByteBufAllocator.java:24)
       io.netty.channel.DefaultChannelConfig.<init>(DefaultChannelConfig.java:60)
       io.netty.channel.DefaultChannelConfig.<init>(DefaultChannelConfig.java:75)
       [...]
     Caused by: java.lang.IllegalArgumentException: Can't find '[toLeakAwareBuffer]' in io.netty.buffer.AbstractByteBufAllocator
       io.netty.util.ResourceLeakDetector.addExclusions(ResourceLeakDetector.java:596)
       io.netty.buffer.AbstractByteBufAllocator.<clinit>(AbstractByteBufAllocator.java:37)
       [...]
  JUnit Jupiter:Netty_codec_socksTest:socks4aCommandRequestSupportsDomainNames()
    MethodSource [className = 'io_netty.netty_codec_socks.Netty_codec_socksTest', methodName = 'socks4aCommandRequestSupportsDomainNames', methodParameterTypes = '']
    => java.lang.NoClassDefFoundError: Could not initialize class io.netty.buffer.ByteBufAllocator
       io.netty.channel.DefaultChannelConfig.<init>(DefaultChannelConfig.java:60)
       io.netty.channel.DefaultChannelConfig.<init>(DefaultChannelConfig.java:75)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:185)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:166)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:150)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:137)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:102)
       io_netty.netty_codec_socks.Netty_codec_socksTest.socks4aCommandRequestSupportsDomainNames(Netty_codec_socksTest.java:100)
       java.base@25.0.3/java.lang.reflect.Method.invoke(Method.java:565)
       org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:727)
       [...]
  JUnit Jupiter:Netty_codec_socksTest:replayingDecodersWaitForCompleteFramesBeforeEmittingMessages()
    MethodSource [className = 'io_netty.netty_codec_socks.Netty_codec_socksTest', methodName = 'replayingDecodersWaitForCompleteFramesBeforeEmittingMessages', methodParameterTypes = '']
    => java.lang.NoClassDefFoundError: Could not initialize class io.netty.buffer.AbstractByteBufAllocator
       java.base@25.0.3/java.lang.Class.ensureInitialized(DynamicHub.java:778)
       io.netty.buffer.Unpooled.<clinit>(Unpooled.java:75)
       io.netty.handler.codec.ReplayingDecoderByteBuf.<clinit>(ReplayingDecoderByteBuf.java:48)
       io.netty.handler.codec.ReplayingDecoder.<init>(ReplayingDecoder.java:272)
       io.netty.handler.codec.socksx.v5.Socks5PasswordAuthRequestDecoder.<init>(Socks5PasswordAuthRequestDecoder.java:46)
       io_netty.netty_codec_socks.Netty_codec_socksTest.replayingDecodersWaitForCompleteFramesBeforeEmittingMessages(Netty_codec_socksTest.java:268)
       java.base@25.0.3/java.lang.reflect.Method.invoke(Method.java:565)
       org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:727)
       org.junit.jupiter.engine.execution.MethodInvocation.proceed(MethodInvocation.java:60)
       org.junit.jupiter.engine.execution.InvocationInterceptorChain$ValidatingInvocation.proceed(InvocationInterceptorChain.java:131)
       [...]
  JUnit Jupiter:Netty_codec_socksTest:socks5CommandCodecsUseCustomAddressCodecs()
    MethodSource [className = 'io_netty.netty_codec_socks.Netty_codec_socksTest', methodName = 'socks5CommandCodecsUseCustomAddressCodecs', methodParameterTypes = '']
    => java.lang.NoClassDefFoundError: Could not initialize class io.netty.buffer.ByteBufAllocator
       io.netty.channel.DefaultChannelConfig.<init>(DefaultChannelConfig.java:60)
       io.netty.channel.DefaultChannelConfig.<init>(DefaultChannelConfig.java:75)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:185)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:166)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:150)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:137)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:102)
       io_netty.netty_codec_socks.Netty_codec_socksTest.socks5CommandCodecsUseCustomAddressCodecs(Netty_codec_socksTest.java:232)
       java.base@25.0.3/java.lang.reflect.Method.invoke(Method.java:565)
       org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:727)
       [...]
  JUnit Jupiter:Netty_codec_socksTest:socks5CommandCodecsSupportIpv4DomainAndIpv6Addresses()
    MethodSource [className = 'io_netty.netty_codec_socks.Netty_codec_socksTest', methodName = 'socks5CommandCodecsSupportIpv4DomainAndIpv6Addresses', methodParameterTypes = '']
    => java.lang.NoClassDefFoundError: Could not initialize class io.netty.buffer.ByteBufAllocator
       io.netty.channel.DefaultChannelConfig.<init>(DefaultChannelConfig.java:60)
       io.netty.channel.DefaultChannelConfig.<init>(DefaultChannelConfig.java:75)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:185)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:166)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:150)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:137)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:102)
       io_netty.netty_codec_socks.Netty_codec_socksTest.assertCommandRequestRoundTrip(Netty_codec_socksTest.java:314)
       io_netty.netty_codec_socks.Netty_codec_socksTest.socks5CommandCodecsSupportIpv4DomainAndIpv6Addresses(Netty_codec_socksTest.java:178)
       java.base@25.0.3/java.lang.reflect.Method.invoke(Method.java:565)
       [...]
  JUnit Jupiter:Netty_codec_socksTest:socks5InitialNegotiationAndPasswordAuthenticationRoundTrip()
    MethodSource [className = 'io_netty.netty_codec_socks.Netty_codec_socksTest', methodName = 'socks5InitialNegotiationAndPasswordAuthenticationRoundTrip', methodParameterTypes = '']
    => java.lang.NoClassDefFoundError: Could not initialize class io.netty.buffer.ByteBufAllocator
       io.netty.channel.DefaultChannelConfig.<init>(DefaultChannelConfig.java:60)
       io.netty.channel.DefaultChannelConfig.<init>(DefaultChannelConfig.java:75)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:185)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:166)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:150)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:137)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:102)
       io_netty.netty_codec_socks.Netty_codec_socksTest.socks5InitialNegotiationAndPasswordAuthenticationRoundTrip(Netty_codec_socksTest.java:123)
       java.base@25.0.3/java.lang.reflect.Method.invoke(Method.java:565)
       org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:727)
       [...]
  JUnit Jupiter:Netty_codec_socksTest:socks4MessagesRoundTripThroughClientAndServerCodecs()

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':nativeTest'.
> Process 'command '/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.netty/netty-codec-socks/4.1.60.Final/build/native/nativeTestCompile/io.netty.netty-codec-socks_tests-tests'' finished with non-zero exit value 1

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

BUILD FAILED in 1m 25s
    MethodSource [className = 'io_netty.netty_codec_socks.Netty_codec_socksTest', methodName = 'socks4MessagesRoundTripThroughClientAndServerCodecs', methodParameterTypes = '']
    => java.lang.NoClassDefFoundError: Could not initialize class io.netty.buffer.ByteBufAllocator
       io.netty.channel.DefaultChannelConfig.<init>(DefaultChannelConfig.java:60)
       io.netty.channel.DefaultChannelConfig.<init>(DefaultChannelConfig.java:75)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:185)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:166)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:150)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:137)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:102)
       io_netty.netty_codec_socks.Netty_codec_socksTest.socks4MessagesRoundTripThroughClientAndServerCodecs(Netty_codec_socksTest.java:60)
       java.base@25.0.3/java.lang.reflect.Method.invoke(Method.java:565)
       org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:727)
       [...]

Test run finished after 6 ms
[         3 containers found      ]
[         0 containers skipped    ]
[         3 containers started    ]
[         0 containers aborted    ]
[         3 containers successful ]
[         0 containers failed     ]
[         8 tests found           ]
[         0 tests skipped         ]
[         8 tests started         ]
[         0 tests aborted         ]
[         1 tests successful      ]
[         7 tests failed          ]


[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.netty/netty-codec-socks/4.1.60.Final/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
16 actionable tasks: 6 executed, 10 up-to-date

> Task :tckTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for io.netty:netty-codec-socks:4.1.60.Final failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 1m 27s
11 actionable tasks: 3 executed, 8 up-to-date
❌ ERROR: 'native-image run' for 4.1.79.Final failed with exit code 1.
FAILED[native-image run][4.1.79.Final][GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="4.1.79.Final" ./gradlew clean test -Pcoordinates="io.netty:netty-codec-socks:4.1.60.Final"]
```

## native-image run on ubuntu-latest / JDK latest-ea / mode current-defaults

Failure kind: native-image run
OS: ubuntu-latest
JDK: latest-ea
Native-image mode: current-defaults
Reproducer: `GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="4.1.79.Final" ./gradlew clean test -Pcoordinates="io.netty:netty-codec-socks:4.1.60.Final"`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/25248087844/job/74035700686
Last 300 lines of the log (excerpt truncated to fit the GitHub issue size limit):
```console
to 'true')
 - 4 thread(s) (100.0% of 4 available processor(s), determined at start)
[junit-platform-native] Running in 'test listener' mode using files matching pattern [junit-platform-unique-ids*] found in folder [/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.netty/netty-codec-socks/4.1.60.Final/build/test-results/test/testlist] and its subfolders.
[2/8] Performing analysis...  [*****]                                                                  (36.7s @ 0.87GiB)
    8,647 types,   9,809 fields, and  38,627 methods found reachable
      319 types,     109 fields, and     712 methods registered for reflection
       60 types,      68 fields, and      52 methods registered for JNI access
      155 resource accesses registered with 462.58KiB total size
        4 native libraries: dl, pthread, rt, z
[3/8] Building universe...                                                                              (4.4s @ 0.97GiB)
[4/8] Parsing methods...      [**]                                                                      (4.3s @ 0.59GiB)
[5/8] Inlining methods...     [***]                                                                     (2.0s @ 0.65GiB)
[6/8] Compiling methods...    [****]                                                                   (20.6s @ 0.70GiB)
[7/8] Laying out methods...   [**]                                                                      (3.2s @ 0.87GiB)
[8/8] Creating image...       [**]                                                                      (3.6s @ 1.00GiB)
   8.72MiB (33.00%) for code area:    23,904 compilation units
  13.88MiB (52.50%) for image heap:  219,882 objects and 25 resources
   3.83MiB (14.50%) for other data
  26.43MiB in total image size, 24.38MiB in total file size
------------------------------------------------------------------------------------------------------------------------
Top 10 origins of code area:                                Top 10 object types in image heap:
 991.35KiB o.g.n.runtime/com.oracle.svm.core.code              2.71MiB byte[] for code metadata
 703.58KiB java.base/java.util                                 2.60MiB byte[] for string data
 408.45KiB java.base/jdk.internal.classfile.impl               1.39MiB java.base/java.lang.String
 322.84KiB java.base/java.lang                                 1.03MiB java.base/java.lang.Class
 244.34KiB java.base/java.util.concurrent                    945.77KiB o.g.n.runtime/c.o.s.core.hub.DynamicHubCompanion
 228.95KiB java.base/java.text                               616.68KiB heap alignment
 224.55KiB java.base/com.sun.crypto.provider                 576.94KiB byte[] for general heap data
 223.93KiB org.assertj.core.internal.bytebuddy.jar.asm       462.98KiB byte[] for embedded resources
 186.77KiB io.netty.buffer                                   430.27KiB byte[] for reflection metadata
 151.95KiB java.base/java.util.stream                        335.35KiB o.g.n.runtime/c.o.s.c.h.ImageReflectionMetadata
   5.12MiB for 325 more packages                               2.85MiB for 2065 more object types
                            Use '--emit build-report' to create a report with more details.
------------------------------------------------------------------------------------------------------------------------
Security report:
 - Binary includes Java deserialization.
 - CycloneDX SBOM with 22 component(s) is embedded in binary (1.12KiB). 1 type(s) could not be associated to a component.
 - Advanced obfuscation not enabled; enable with '-H:AdvancedObfuscation=""' (experimental support).
------------------------------------------------------------------------------------------------------------------------
Recommendations:
 G1GC: Use the G1 GC ('--gc=G1') for improved latency and throughput.
 PGO:  Use Profile-Guided Optimizations ('--pgo') for improved throughput.
 FUTR: Use '--future-defaults=all' to prepare for future releases.
 HEAP: Set max heap for improved and more predictable memory usage.
 CPU:  Enable more CPU features with '-march=native' for improved performance.
------------------------------------------------------------------------------------------------------------------------
                       5.5s (6.4% of total time) in 721 GCs | Peak RSS: 1.62GiB | CPU load: 3.60
------------------------------------------------------------------------------------------------------------------------
Build artifacts:
 /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.netty/netty-codec-socks/4.1.60.Final/build/native/nativeTestCompile/io.netty.netty-codec-socks_tests-tests (executable, 24.38MiB)
 /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.netty/netty-codec-socks/4.1.60.Final/build/native/nativeTestCompile/libmanagement_ext.so (jdk_library, 32.23KiB)
 /tmp/native-image-build-output.json (build_info, 1.03KiB)
========================================================================================================================
Finished generating 'io.netty.netty-codec-socks_tests-tests' in 1m 24s.

The build process encountered 5 warnings.

[native-image-plugin] Native Image written to: /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.netty/netty-codec-socks/4.1.60.Final/build/native/nativeTestCompile

> Task :nativeTest FAILED
[junit-platform-native] Running in 'test listener' mode using files matching pattern [junit-platform-unique-ids*] found in folder [/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.netty/netty-codec-socks/4.1.60.Final/build/test-results/test/testlist] and its subfolders.
JUnit Platform on Native Image - report
----------------------------------------

io_netty.netty_codec_socks.Netty_codec_socksTest > portUnificationServerHandlerSelectsSocks4AndSocks5Decoders() FAILED

io_netty.netty_codec_socks.Netty_codec_socksTest > socks4aCommandRequestSupportsDomainNames() FAILED

io_netty.netty_codec_socks.Netty_codec_socksTest > protocolConstantsAndExtensibleValueObjectsExposeWireValues() SUCCESSFUL

io_netty.netty_codec_socks.Netty_codec_socksTest > replayingDecodersWaitForCompleteFramesBeforeEmittingMessages() FAILED

io_netty.netty_codec_socks.Netty_codec_socksTest > socks5CommandCodecsUseCustomAddressCodecs() FAILED

io_netty.netty_codec_socks.Netty_codec_socksTest > socks5CommandCodecsSupportIpv4DomainAndIpv6Addresses() FAILED

io_netty.netty_codec_socks.Netty_codec_socksTest > socks5InitialNegotiationAndPasswordAuthenticationRoundTrip() FAILED

io_netty.netty_codec_socks.Netty_codec_socksTest > socks4MessagesRoundTripThroughClientAndServerCodecs() FAILED


Failures (7):
  JUnit Jupiter:Netty_codec_socksTest:portUnificationServerHandlerSelectsSocks4AndSocks5Decoders()
    MethodSource [className = 'io_netty.netty_codec_socks.Netty_codec_socksTest', methodName = 'portUnificationServerHandlerSelectsSocks4AndSocks5Decoders', methodParameterTypes = '']
    => java.lang.ExceptionInInitializerError
       io.netty.buffer.ByteBufUtil.<clinit>(ByteBufUtil.java:86)
       io.netty.buffer.ByteBufAllocator.<clinit>(ByteBufAllocator.java:24)
       io.netty.channel.DefaultChannelConfig.<init>(DefaultChannelConfig.java:60)
       io.netty.channel.DefaultChannelConfig.<init>(DefaultChannelConfig.java:75)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:185)
       [...]
     Caused by: java.lang.IllegalArgumentException: Can't find '[toLeakAwareBuffer]' in io.netty.buffer.AbstractByteBufAllocator
       io.netty.util.ResourceLeakDetector.addExclusions(ResourceLeakDetector.java:596)
       io.netty.buffer.AbstractByteBufAllocator.<clinit>(AbstractByteBufAllocator.java:37)
       [...]
  JUnit Jupiter:Netty_codec_socksTest:socks4aCommandRequestSupportsDomainNames()
    MethodSource [className = 'io_netty.netty_codec_socks.Netty_codec_socksTest', methodName = 'socks4aCommandRequestSupportsDomainNames', methodParameterTypes = '']
    => java.lang.NoClassDefFoundError: Could not initialize class io.netty.buffer.ByteBufAllocator
       io.netty.channel.DefaultChannelConfig.<init>(DefaultChannelConfig.java:60)
       io.netty.channel.DefaultChannelConfig.<init>(DefaultChannelConfig.java:75)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:185)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:166)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:150)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:137)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:102)
       io_netty.netty_codec_socks.Netty_codec_socksTest.socks4aCommandRequestSupportsDomainNames(Netty_codec_socksTest.java:100)
       java.base@25.0.2/java.lang.reflect.Method.invoke(Method.java:565)
       org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:727)
       [...]
  JUnit Jupiter:Netty_codec_socksTest:replayingDecodersWaitForCompleteFramesBeforeEmittingMessages()
    MethodSource [className = 'io_netty.netty_codec_socks.Netty_codec_socksTest', methodName = 'replayingDecodersWaitForCompleteFramesBeforeEmittingMessages', methodParameterTypes = '']
    => java.lang.NoClassDefFoundError: Could not initialize class io.netty.buffer.AbstractByteBufAllocator
       io.netty.buffer.Unpooled.<clinit>(Unpooled.java:75)
       io.netty.handler.codec.ReplayingDecoderByteBuf.<clinit>(ReplayingDecoderByteBuf.java:48)
       io.netty.handler.codec.ReplayingDecoder.<init>(ReplayingDecoder.java:272)
       io.netty.handler.codec.socksx.v5.Socks5PasswordAuthRequestDecoder.<init>(Socks5PasswordAuthRequestDecoder.java:46)
       io_netty.netty_codec_socks.Netty_codec_socksTest.replayingDecodersWaitForCompleteFramesBeforeEmittingMessages(Netty_codec_socksTest.java:268)
       java.base@25.0.2/java.lang.reflect.Method.invoke(Method.java:565)
       org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:727)
       org.junit.jupiter.engine.execution.MethodInvocation.proceed(MethodInvocation.java:60)
       org.junit.jupiter.engine.execution.InvocationInterceptorChain$ValidatingInvocation.proceed(InvocationInterceptorChain.java:131)
       org.junit.jupiter.engine.extension.TimeoutExtension.intercept(TimeoutExtension.java:156)
       [...]
  JUnit Jupiter:Netty_codec_socksTest:socks5CommandCodecsUseCustomAddressCodecs()
    MethodSource [className = 'io_netty.netty_codec_socks.Netty_codec_socksTest', methodName = 'socks5CommandCodecsUseCustomAddressCodecs', methodParameterTypes = '']
    => java.lang.NoClassDefFoundError: Could not initialize class io.netty.buffer.ByteBufAllocator
       io.netty.channel.DefaultChannelConfig.<init>(DefaultChannelConfig.java:60)
       io.netty.channel.DefaultChannelConfig.<init>(DefaultChannelConfig.java:75)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:185)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:166)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:150)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:137)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:102)
       io_netty.netty_codec_socks.Netty_codec_socksTest.socks5CommandCodecsUseCustomAddressCodecs(Netty_codec_socksTest.java:232)
       java.base@25.0.2/java.lang.reflect.Method.invoke(Method.java:565)
       org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:727)
       [...]
  JUnit Jupiter:Netty_codec_socksTest:socks5CommandCodecsSupportIpv4DomainAndIpv6Addresses()
    MethodSource [className = 'io_netty.netty_codec_socks.Netty_codec_socksTest', methodName = 'socks5CommandCodecsSupportIpv4DomainAndIpv6Addresses', methodParameterTypes = '']
    => java.lang.NoClassDefFoundError: Could not initialize class io.netty.buffer.ByteBufAllocator
       io.netty.channel.DefaultChannelConfig.<init>(DefaultChannelConfig.java:60)
       io.netty.channel.DefaultChannelConfig.<init>(DefaultChannelConfig.java:75)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:185)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:166)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:150)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:137)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:102)
       io_netty.netty_codec_socks.Netty_codec_socksTest.assertCommandRequestRoundTrip(Netty_codec_socksTest.java:314)
       io_netty.netty_codec_socks.Netty_codec_socksTest.socks5CommandCodecsSupportIpv4DomainAndIpv6Addresses(Netty_codec_socksTest.java:178)
       java.base@25.0.2/java.lang.reflect.Method.invoke(Method.java:565)
       [...]
  JUnit Jupiter:Netty_codec_socksTest:socks5InitialNegotiationAndPasswordAuthenticationRoundTrip()
    MethodSource [className = 'io_netty.netty_codec_socks.Netty_codec_socksTest', methodName = 'socks5InitialNegotiationAndPasswordAuthenticationRoundTrip', methodParameterTypes = '']
    => java.lang.NoClassDefFoundError: Could not initialize class io.netty.buffer.ByteBufAllocator
       io.netty.channel.DefaultChannelConfig.<init>(DefaultChannelConfig.java:60)
       io.netty.channel.DefaultChannelConfig.<init>(DefaultChannelConfig.java:75)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:185)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:166)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:150)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:137)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:102)
       io_netty.netty_codec_socks.Netty_codec_socksTest.socks5InitialNegotiationAndPasswordAuthenticationRoundTrip(Netty_codec_socksTest.java:123)
       java.base@25.0.2/java.lang.reflect.Method.invoke(Method.java:565)
       org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:727)
       [...]
  JUnit Jupiter:Netty_codec_socksTest:socks4MessagesRoundTripThroughClientAndServerCodecs()
    MethodSource [className = 'io_netty.netty_codec_socks.Netty_codec_socksTest', methodName = 'socks4MessagesRoundTripThroughClientAndServerCodecs', methodParameterTypes = '']
    => java.lang.NoClassDefFoundError: Could not initialize class io.netty.buffer.ByteBufAllocator
       io.netty.channel.DefaultChannelConfig.<init>(DefaultChannelConfig.java:60)
       io.netty.channel.DefaultChannelConfig.<init>(DefaultChannelConfig.java:75)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:185)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:166)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:150)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:137)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:102)
       io_netty.netty_codec_socks.Netty_codec_socksTest.socks4MessagesRoundTripThroughClientAndServerCodecs(Netty_codec_socksTest.java:60)
       java.base@25.0.2/java.lang.reflect.Method.invoke(Method.java:565)
       org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:727)
       [...]

Test run finished after 6 ms
[         3 containers found      ]
[         0 containers skipped    ]
[         3 containers started    ]
[         0 containers aborted    ]
[         3 containers successful ]
[         0 containers failed     ]
[         8 tests found           ]
[         0 tests skipped         ]
[         8 tests started         ]
[         0 tests aborted         ]
[         1 tests successful      ]
[         7 tests failed          ]


[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.netty/netty-codec-socks/4.1.60.Final/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
16 actionable tasks: 6 executed, 10 up-to-date
WARNING: A terminally deprecated method in sun.misc.Unsafe has been called
WARNING: sun.misc.Unsafe::getLong has been called by io.netty.util.internal.PlatformDependent0$4 (file:/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.netty/netty-codec-socks/4.1.60.Final/build/native/nativeTestCompile/io.netty.netty-codec-socks_tests-tests)
WARNING: Please consider reporting this to the maintainers of class io.netty.util.internal.PlatformDependent0$4
WARNING: sun.misc.Unsafe::getLong will be removed in a future release

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':nativeTest'.
> Process 'command '/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.netty/netty-codec-socks/4.1.60.Final/build/native/nativeTestCompile/io.netty.netty-codec-socks_tests-tests'' finished with non-zero exit value 1

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

BUILD FAILED in 1m 28s

> Task :tckTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for io.netty:netty-codec-socks:4.1.60.Final failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 1m 30s
11 actionable tasks: 3 executed, 8 up-to-date
❌ ERROR: 'native-image run' for 4.1.79.Final failed with exit code 1.
FAILED[native-image run][4.1.79.Final][GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="4.1.79.Final" ./gradlew clean test -Pcoordinates="io.netty:netty-codec-socks:4.1.60.Final"]
```

## native-image run on ubuntu-latest / JDK latest-ea / mode future-defaults-all

Failure kind: native-image run
OS: ubuntu-latest
JDK: latest-ea
Native-image mode: future-defaults-all
Reproducer: `GVM_TCK_NATIVE_IMAGE_MODE="future-defaults-all" GVM_TCK_LV="4.1.79.Final" ./gradlew clean test -Pcoordinates="io.netty:netty-codec-socks:4.1.60.Final"`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/25248087844/job/74035700674
Last 300 lines of the log (excerpt truncated to fit the GitHub issue size limit):
```console
 (85% of system memory because $CI set to 'true')
 - 4 thread(s) (100.0% of 4 available processor(s), determined at start)
[junit-platform-native] Running in 'test listener' mode using files matching pattern [junit-platform-unique-ids*] found in folder [/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.netty/netty-codec-socks/4.1.60.Final/build/test-results/test/testlist] and its subfolders.
[2/8] Performing analysis...  [*****]                                                                  (33.1s @ 0.95GiB)
    9,187 types,  10,611 fields, and  41,659 methods found reachable
      447 types,     109 fields, and     918 methods registered for reflection
       60 types,      68 fields, and      52 methods registered for JNI access
      157 resource accesses registered with 462.72KiB total size
        4 native libraries: dl, pthread, rt, z
[3/8] Building universe...                                                                              (4.0s @ 1.07GiB)
[4/8] Parsing methods...      [**]                                                                      (4.0s @ 0.78GiB)
[5/8] Inlining methods...     [***]                                                                     (1.8s @ 0.84GiB)
[6/8] Compiling methods...    [****]                                                                   (19.4s @ 0.85GiB)
[7/8] Laying out methods...   [**]                                                                      (3.2s @ 1.03GiB)
[8/8] Creating image...       [**]                                                                      (3.7s @ 1.19GiB)
   9.84MiB (33.18%) for code area:    26,216 compilation units
  15.69MiB (52.88%) for image heap:  258,814 objects and 27 resources
   4.13MiB (13.94%) for other data
  29.66MiB in total image size, 27.44MiB in total file size
------------------------------------------------------------------------------------------------------------------------
Top 10 origins of code area:                                Top 10 object types in image heap:
   1.10MiB o.g.n.runtime/com.oracle.svm.core.code              3.07MiB byte[] for code metadata
 710.74KiB java.base/java.util                                 2.89MiB byte[] for string data
 408.42KiB java.base/jdk.internal.classfile.impl               1.53MiB java.base/java.lang.String
 332.47KiB java.base/java.lang                                 1.11MiB java.base/java.lang.Class
 267.01KiB java.base/com.sun.crypto.provider                   0.98MiB o.g.n.runtime/c.o.s.core.hub.DynamicHubCompanion
 244.45KiB java.base/java.util.concurrent                    925.50KiB byte[] for general heap data
 231.52KiB java.base/java.text                               498.42KiB byte[] for reflection metadata
 223.93KiB org.assertj.core.internal.bytebuddy.jar.asm       463.16KiB byte[] for embedded resources
 194.34KiB java.base/sun.security.provider                   363.30KiB java.base/java.lang.Object[]
 186.73KiB io.netty.buffer                                   356.45KiB o.g.n.runtime/c.o.s.c.h.ImageReflectionMetadata
   6.01MiB for 335 more packages                               3.57MiB for 2220 more object types
                            Use '--emit build-report' to create a report with more details.
------------------------------------------------------------------------------------------------------------------------
Security report:
 - Binary includes Java deserialization.
 - CycloneDX SBOM with 22 component(s) is embedded in binary (1.12KiB). 1 type(s) could not be associated to a component.
 - Advanced obfuscation not enabled; enable with '-H:AdvancedObfuscation=""' (experimental support).
------------------------------------------------------------------------------------------------------------------------
Recommendations:
 G1GC: Use the G1 GC ('--gc=G1') for improved latency and throughput.
 PGO:  Use Profile-Guided Optimizations ('--pgo') for improved throughput.
 HEAP: Set max heap for improved and more predictable memory usage.
 CPU:  Enable more CPU features with '-march=native' for improved performance.
------------------------------------------------------------------------------------------------------------------------
                       6.3s (8.0% of total time) in 774 GCs | Peak RSS: 1.83GiB | CPU load: 3.61
------------------------------------------------------------------------------------------------------------------------
Build artifacts:
 /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.netty/netty-codec-socks/4.1.60.Final/build/native/nativeTestCompile/io.netty.netty-codec-socks_tests-tests (executable, 27.44MiB)
 /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.netty/netty-codec-socks/4.1.60.Final/build/native/nativeTestCompile/libmanagement_ext.so (jdk_library, 32.23KiB)
 /tmp/native-image-build-output.json (build_info, 1.03KiB)
========================================================================================================================
Finished generating 'io.netty.netty-codec-socks_tests-tests' in 1m 17s.

The build process encountered 5 warnings.

[native-image-plugin] Native Image written to: /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.netty/netty-codec-socks/4.1.60.Final/build/native/nativeTestCompile

> Task :nativeTest FAILED
[junit-platform-native] Running in 'test listener' mode using files matching pattern [junit-platform-unique-ids*] found in folder [/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.netty/netty-codec-socks/4.1.60.Final/build/test-results/test/testlist] and its subfolders.
JUnit Platform on Native Image - report
----------------------------------------

io_netty.netty_codec_socks.Netty_codec_socksTest > portUnificationServerHandlerSelectsSocks4AndSocks5Decoders() FAILED

io_netty.netty_codec_socks.Netty_codec_socksTest > socks4aCommandRequestSupportsDomainNames() FAILED

io_netty.netty_codec_socks.Netty_codec_socksTest > protocolConstantsAndExtensibleValueObjectsExposeWireValues() SUCCESSFUL

io_netty.netty_codec_socks.Netty_codec_socksTest > replayingDecodersWaitForCompleteFramesBeforeEmittingMessages() FAILED

io_netty.netty_codec_socks.Netty_codec_socksTest > socks5CommandCodecsUseCustomAddressCodecs() FAILED

io_netty.netty_codec_socks.Netty_codec_socksTest > socks5CommandCodecsSupportIpv4DomainAndIpv6Addresses() FAILED

io_netty.netty_codec_socks.Netty_codec_socksTest > socks5InitialNegotiationAndPasswordAuthenticationRoundTrip() FAILED

io_netty.netty_codec_socks.Netty_codec_socksTest > socks4MessagesRoundTripThroughClientAndServerCodecs() FAILED


Failures (7):
  JUnit Jupiter:Netty_codec_socksTest:portUnificationServerHandlerSelectsSocks4AndSocks5Decoders()
    MethodSource [className = 'io_netty.netty_codec_socks.Netty_codec_socksTest', methodName = 'portUnificationServerHandlerSelectsSocks4AndSocks5Decoders', methodParameterTypes = '']
    => java.lang.ExceptionInInitializerError
       io.netty.buffer.ByteBufUtil.<clinit>(ByteBufUtil.java:86)
       io.netty.buffer.ByteBufAllocator.<clinit>(ByteBufAllocator.java:24)
       io.netty.channel.DefaultChannelConfig.<init>(DefaultChannelConfig.java:60)
       io.netty.channel.DefaultChannelConfig.<init>(DefaultChannelConfig.java:75)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:185)
       [...]
     Caused by: java.lang.IllegalArgumentException: Can't find '[toLeakAwareBuffer]' in io.netty.buffer.AbstractByteBufAllocator
       io.netty.util.ResourceLeakDetector.addExclusions(ResourceLeakDetector.java:596)
       io.netty.buffer.AbstractByteBufAllocator.<clinit>(AbstractByteBufAllocator.java:37)
       [...]
  JUnit Jupiter:Netty_codec_socksTest:socks4aCommandRequestSupportsDomainNames()
    MethodSource [className = 'io_netty.netty_codec_socks.Netty_codec_socksTest', methodName = 'socks4aCommandRequestSupportsDomainNames', methodParameterTypes = '']
    => java.lang.NoClassDefFoundError: Could not initialize class io.netty.buffer.ByteBufAllocator
       io.netty.channel.DefaultChannelConfig.<init>(DefaultChannelConfig.java:60)
       io.netty.channel.DefaultChannelConfig.<init>(DefaultChannelConfig.java:75)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:185)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:166)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:150)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:137)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:102)
       io_netty.netty_codec_socks.Netty_codec_socksTest.socks4aCommandRequestSupportsDomainNames(Netty_codec_socksTest.java:100)
       java.base@25.0.2/java.lang.reflect.Method.invoke(Method.java:565)
       org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:727)
       [...]
  JUnit Jupiter:Netty_codec_socksTest:replayingDecodersWaitForCompleteFramesBeforeEmittingMessages()
    MethodSource [className = 'io_netty.netty_codec_socks.Netty_codec_socksTest', methodName = 'replayingDecodersWaitForCompleteFramesBeforeEmittingMessages', methodParameterTypes = '']
    => java.lang.NoClassDefFoundError: Could not initialize class io.netty.buffer.AbstractByteBufAllocator
       io.netty.buffer.Unpooled.<clinit>(Unpooled.java:75)
       io.netty.handler.codec.ReplayingDecoderByteBuf.<clinit>(ReplayingDecoderByteBuf.java:48)
       io.netty.handler.codec.ReplayingDecoder.<init>(ReplayingDecoder.java:272)
       io.netty.handler.codec.socksx.v5.Socks5PasswordAuthRequestDecoder.<init>(Socks5PasswordAuthRequestDecoder.java:46)
       io_netty.netty_codec_socks.Netty_codec_socksTest.replayingDecodersWaitForCompleteFramesBeforeEmittingMessages(Netty_codec_socksTest.java:268)
       java.base@25.0.2/java.lang.reflect.Method.invoke(Method.java:565)
       org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:727)
       org.junit.jupiter.engine.execution.MethodInvocation.proceed(MethodInvocation.java:60)
       org.junit.jupiter.engine.execution.InvocationInterceptorChain$ValidatingInvocation.proceed(InvocationInterceptorChain.java:131)
       org.junit.jupiter.engine.extension.TimeoutExtension.intercept(TimeoutExtension.java:156)
       [...]
  JUnit Jupiter:Netty_codec_socksTest:socks5CommandCodecsUseCustomAddressCodecs()
    MethodSource [className = 'io_netty.netty_codec_socks.Netty_codec_socksTest', methodName = 'socks5CommandCodecsUseCustomAddressCodecs', methodParameterTypes = '']
    => java.lang.NoClassDefFoundError: Could not initialize class io.netty.buffer.ByteBufAllocator
       io.netty.channel.DefaultChannelConfig.<init>(DefaultChannelConfig.java:60)
       io.netty.channel.DefaultChannelConfig.<init>(DefaultChannelConfig.java:75)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:185)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:166)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:150)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:137)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:102)
       io_netty.netty_codec_socks.Netty_codec_socksTest.socks5CommandCodecsUseCustomAddressCodecs(Netty_codec_socksTest.java:232)
       java.base@25.0.2/java.lang.reflect.Method.invoke(Method.java:565)
       org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:727)
       [...]
  JUnit Jupiter:Netty_codec_socksTest:socks5CommandCodecsSupportIpv4DomainAndIpv6Addresses()
    MethodSource [className = 'io_netty.netty_codec_socks.Netty_codec_socksTest', methodName = 'socks5CommandCodecsSupportIpv4DomainAndIpv6Addresses', methodParameterTypes = '']
    => java.lang.NoClassDefFoundError: Could not initialize class io.netty.buffer.ByteBufAllocator
       io.netty.channel.DefaultChannelConfig.<init>(DefaultChannelConfig.java:60)
       io.netty.channel.DefaultChannelConfig.<init>(DefaultChannelConfig.java:75)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:185)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:166)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:150)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:137)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:102)
       io_netty.netty_codec_socks.Netty_codec_socksTest.assertCommandRequestRoundTrip(Netty_codec_socksTest.java:314)
       io_netty.netty_codec_socks.Netty_codec_socksTest.socks5CommandCodecsSupportIpv4DomainAndIpv6Addresses(Netty_codec_socksTest.java:178)
       java.base@25.0.2/java.lang.reflect.Method.invoke(Method.java:565)
       [...]
  JUnit Jupiter:Netty_codec_socksTest:socks5InitialNegotiationAndPasswordAuthenticationRoundTrip()
    MethodSource [className = 'io_netty.netty_codec_socks.Netty_codec_socksTest', methodName = 'socks5InitialNegotiationAndPasswordAuthenticationRoundTrip', methodParameterTypes = '']
    => java.lang.NoClassDefFoundError: Could not initialize class io.netty.buffer.ByteBufAllocator
       io.netty.channel.DefaultChannelConfig.<init>(DefaultChannelConfig.java:60)
       io.netty.channel.DefaultChannelConfig.<init>(DefaultChannelConfig.java:75)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:185)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:166)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:150)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:137)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:102)
       io_netty.netty_codec_socks.Netty_codec_socksTest.socks5InitialNegotiationAndPasswordAuthenticationRoundTrip(Netty_codec_socksTest.java:123)
       java.base@25.0.2/java.lang.reflect.Method.invoke(Method.java:565)
       org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:727)
       [...]
  JUnit Jupiter:Netty_codec_socksTest:socks4MessagesRoundTripThroughClientAndServerCodecs()
    MethodSource [className = 'io_netty.netty_codec_socks.Netty_codec_socksTest', methodName = 'socks4MessagesRoundTripThroughClientAndServerCodecs', methodParameterTypes = '']
    => java.lang.NoClassDefFoundError: Could not initialize class io.netty.buffer.ByteBufAllocator
       io.netty.channel.DefaultChannelConfig.<init>(DefaultChannelConfig.java:60)
       io.netty.channel.DefaultChannelConfig.<init>(DefaultChannelConfig.java:75)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:185)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:166)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:150)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:137)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:102)
       io_netty.netty_codec_socks.Netty_codec_socksTest.socks4MessagesRoundTripThroughClientAndServerCodecs(Netty_codec_socksTest.java:60)
       java.base@25.0.2/java.lang.reflect.Method.invoke(Method.java:565)
       org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:727)
       [...]

Test run finished after 6 ms
[         3 containers found      ]
[         0 containers skipped    ]
[         3 containers started    ]
[         0 containers aborted    ]
[         3 containers successful ]
[         0 containers failed     ]
[         8 tests found           ]
[         0 tests skipped         ]
[         8 tests started         ]
[         0 tests aborted         ]
[         1 tests successful      ]
[         7 tests failed          ]


[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.netty/netty-codec-socks/4.1.60.Final/build/reports/problems/problems-report.html
WARNING: A terminally deprecated method in sun.misc.Unsafe has been called
WARNING: sun.misc.Unsafe::getLong has been called by io.netty.util.internal.PlatformDependent0$4 (file:/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.netty/netty-codec-socks/4.1.60.Final/build/native/nativeTestCompile/io.netty.netty-codec-socks_tests-tests)
WARNING: Please consider reporting this to the maintainers of class io.netty.util.internal.PlatformDependent0$4
WARNING: sun.misc.Unsafe::getLong will be removed in a future release

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':nativeTest'.
> Process 'command '/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.netty/netty-codec-socks/4.1.60.Final/build/native/nativeTestCompile/io.netty.netty-codec-socks_tests-tests'' finished with non-zero exit value 1

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 1m 21s
16 actionable tasks: 6 executed, 10 up-to-date

> Task :tckTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for io.netty:netty-codec-socks:4.1.60.Final failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 1m 23s
11 actionable tasks: 3 executed, 8 up-to-date
❌ ERROR: 'native-image run' for 4.1.79.Final failed with exit code 1.
FAILED[native-image run][4.1.79.Final][GVM_TCK_NATIVE_IMAGE_MODE="future-defaults-all" GVM_TCK_LV="4.1.79.Final" ./gradlew clean test -Pcoordinates="io.netty:netty-codec-socks:4.1.60.Final"]
```

--- a/metadata/io.netty/netty-codec-socks/4.1.79.Final/reachability-metadata.json
+++ b/metadata/io.netty/netty-codec-socks/4.1.79.Final/reachability-metadata.json
@@ -1,0 +1,75 @@
+{
+  "reflection": [
+    {
+      "condition": {
+        "typeReached": "io.netty.buffer.AbstractByteBufAllocator"
+      },
+      "type": "io.netty.buffer.AbstractByteBufAllocator",
+      "methods": [
+        {
+          "name": "toLeakAwareBuffer",
+          "parameterTypes": [
+            "io.netty.buffer.ByteBuf"
+          ]
+        },
+        {
+          "name": "toLeakAwareBuffer",
+          "parameterTypes": [
+            "io.netty.buffer.CompositeByteBuf"
+          ]
+        }
+      ]
+    },
+    {
+      "condition": {
+        "typeReached": "io.netty.handler.codec.MessageToByteEncoder"
+      },
+      "type": "io.netty.buffer.AbstractReferenceCountedByteBuf",
+      "fields": [
+        {
+          "name": "refCnt"
+        }
+      ]
+    },
+    {
+      "condition": {
+        "typeReached": "io.netty.handler.codec.socksx.SocksPortUnificationServerHandler"
+      },
+      "type": "io.netty.handler.codec.socksx.v4.Socks4ServerDecoder"
+    },
+    {
+      "condition": {
+        "typeReached": "io.netty.handler.codec.socksx.SocksPortUnificationServerHandler"
+      },
+      "type": "io.netty.handler.codec.socksx.v4.Socks4ServerEncoder"
+    },
+    {
+      "condition": {
+        "typeReached": "io.netty.handler.codec.socksx.SocksPortUnificationServerHandler"
+      },
+      "type": "io.netty.handler.codec.socksx.v5.Socks5InitialRequestDecoder"
+    },
+    {
+      "condition": {
+        "typeReached": "io.netty.handler.codec.socksx.SocksPortUnificationServerHandler"
+      },
+      "type": "io.netty.handler.codec.socksx.v5.Socks5ServerEncoder"
+    }
+  ],
+  "resources": [
+    {
+      "condition": {
+        "typeReached": "io.netty.handler.codec.socksx.v4.DefaultSocks4CommandRequest"
+      },
+      "module": "java.base",
+      "glob": "jdk/internal/icu/impl/data/icudt76b/uprops.icu"
+    },
+    {
+      "condition": {
+        "typeReached": "io.netty.handler.codec.socksx.v4.DefaultSocks4CommandRequest"
+      },
+      "module": "java.base",
+      "glob": "sun/net/idn/uidna.spp"
+    }
+  ]
+}

--- a/metadata/io.netty/netty-codec-socks/index.json
+++ b/metadata/io.netty/netty-codec-socks/index.json
@@ -1,6 +1,22 @@
 [
   {
     "latest" : true,
+    "metadata-version" : "4.1.79.Final",
+    "source-code-url" : "https://repo1.maven.org/maven2/io/netty/netty-codec-socks/4.1.79.Final/netty-codec-socks-4.1.79.Final-sources.jar",
+    "repository-url" : "https://github.com/netty/netty",
+    "test-code-url" : "https://repo1.maven.org/maven2/io/netty/netty-codec-socks/4.1.79.Final/netty-codec-socks-4.1.79.Final-test-sources.jar",
+    "documentation-url" : "https://repo1.maven.org/maven2/io/netty/netty-codec-socks/4.1.79.Final/netty-codec-socks-4.1.79.Final-javadoc.jar",
+    "description" : "Netty Codec SOCKS provides encoders, decoders, and message types for SOCKS protocol support in Netty pipelines. It lets Netty applications implement SOCKS client or server handshakes and proxy-related messaging on top of Netty's asynchronous networking framework.",
+    "test-version" : "4.1.60.Final",
+    "tested-versions" : [
+      "4.1.79.Final"
+    ],
+    "allowed-packages" : [
+      "io.netty.buffer",
+      "io.netty.handler.codec"
+    ]
+  },
+  {
     "metadata-version" : "4.1.60.Final",
     "source-code-url" : "https://repo1.maven.org/maven2/io/netty/netty-codec-socks/4.1.60.Final/netty-codec-socks-4.1.60.Final-sources.jar",
     "repository-url" : "https://github.com/netty/netty",

--- a/stats/io.netty/netty-codec-socks/4.1.79.Final/stats.json
+++ b/stats/io.netty/netty-codec-socks/4.1.79.Final/stats.json
@@ -1,0 +1,31 @@
+{
+  "versions" : [ {
+    "version" : "4.1.79.Final",
+    "dynamicAccess" : {
+      "breakdown" : { },
+      "coverageRatio" : 1.0,
+      "coveredCalls" : 0,
+      "totalCalls" : 0
+    },
+    "libraryCoverage" : {
+      "instruction" : {
+        "covered" : 2194,
+        "missed" : 3952,
+        "ratio" : 0.35698,
+        "total" : 6146
+      },
+      "line" : {
+        "covered" : 518,
+        "missed" : 784,
+        "ratio" : 0.397849,
+        "total" : 1302
+      },
+      "method" : {
+        "covered" : 133,
+        "missed" : 154,
+        "ratio" : 0.463415,
+        "total" : 287
+      }
+    }
+  } ]
+}

